### ISSUE TITLE
Assert batch_size==1 error when batch_size is higher than 1

### ISSUE THREAD

**Issue by henrywotton** [dptech-corp/Uni-Core#18]
Hi,

Thanks for developing this powerful package. When I trained a programme using a batch_size higher than 1, Uni-Core will assert whether the batch_seize is equal to 1 and return AssertionError. Is this normal or it will affect downstream processes?

**Comment by guolinke** [comment]
can you provide the settings? If you use the per-sample gradient clipping, the batch size is limited to 1.
